sashimi-webapp/src/database/data-modifier/dataAdd.js

```diff
@@ -91,6 +93,28 @@ export default class dataAdd {
     );
   }
 
+  static duplicateFile(fileId) {
+    return new Promise((resolve, reject) => {
+      sqlCommands.getMaxFileId()
+      .then((maxId) => {
+        const newFileId = maxId + 1;
+        return sqlCommands.retrieveFullFile(fileId)
+        .then((file) => {
+          file[0].file_id = newFileId;
+          return file;
+        })
+        .then(duplicatedFile =>
+          sqlCommands.insertContent(constants.ENTITIES_FILE_MANAGER, duplicatedFile)
+          .then(() =>
+            resolve(duplicatedFile)
+          )
+          .catch(sqlError => reject(sqlError)))
+        .catch(sqlError => reject(sqlError));
+      })
+      .catch(sqlError => reject(sqlError));
+    });
+  }
+
```

Comment:
is it possible to flatten the promises chain by using `.all` here.

Example:
```javascript
static duplicateFile(fileId) {
  return Promise.all([
    sqlCommands.getMaxFileId(),
    sqlCommands.retrieveFullFile(fileId)
  ])
  .then(result => {
    const maxId = result[0];
    const file = result[1];
    const newFileId = maxId + 1;
    file[0].file_id = newFileId;
    return file
  })
  .then((duplicatedFile) => {
    return sqlCommands.insertContent(constants.ENTITIES_FILE_MANAGER, duplicatedFile)
    .then(() => duplicatedFile);
  });
}
```

Actually one of the benefit of using Promise for handling async code is so that we can minimise nesting of functions.
See: http://www.datchley.name/promise-patterns-anti-patterns/#2nestedpromises